qiita_ware/processing_pipeline.py

```diff
@@ -158,13 +187,68 @@ def _get_preprocess_fastq_cmd(raw_data, prep_template, params):
     forward_seqs = sorted(forward_seqs)
     reverse_seqs = sorted(reverse_seqs)
     barcode_fps = sorted(barcode_fps)
-    mapping_fps = sorted(mapping_fps)
 
-    # Create the split_libraries_fastq.py command
-    cmd = str("split_libraries_fastq.py --store_demultiplexed_fastq -i %s -b "
-              "%s -m %s -o %s %s"
-              % (','.join(forward_seqs), ','.join(barcode_fps),
-                 ','.join(mapping_fps), output_dir, params_str))
+    # Create a temporary directory to store the split libraries output
+    output_dir = mkdtemp(dir=qiita_config.working_dir, prefix='slq_out')
+
+    # Add any other parameter needed to split libraries fastq
+    params_str = params.to_str()
+
+    if len(forward_seqs) == 0:
+        raise ValueError("Forward reads file not found on raw data %s"
+                         % raw_data.id)
+
+    if filetype == "per_sample_FASTQ":
+        if barcode_fps:
+            raise ValueError("per_sample_FASTQ can not have barcodes:"
+                             % (', '.join(barcode_fps)))
+
+        sn_by_rp = _get_sample_names_by_run_prefix(prep_template)
+        samples = []
+        for f in forward_seqs:
+            # getting just the main filename
+            f = basename(f).split('_', 1)[1]
+            # removing extentions: fastq or fastq.gz
+            if f.endswith('fastq.gz'):
+                f = f[:-9]
+            else:
+                f = f[:-6]
+            try:
+                samples.append(sn_by_rp[f])
+                del sn_by_rp[f]
+            except KeyError:
```

Comment:
since the error is ignored, why not just do an explicit check? If the `KeyError` is common, this will be very expensive.

```python
samples.append(sn_by_rp.pop(f)) if f in sn_by_rp else None
```